_LowLevelCode/cpp/test/mex_bin_plugin.tests/mex_bin_plugin.test.cpp

```diff
@@ -0,0 +1,168 @@
+#include "mex_bin_plugin/bin_io_handler.h"
+#include "utility/environment.h"
+#include <gtest/gtest.h>
+
+using namespace Horace::Utility;
+
+bool file_exists(const  std::string& filename) {
+    std::ifstream f(filename.c_str());
+    return f.good();
+};
+
+void del_file(const std::string& filename) {
+    std::remove(filename.c_str());
+};
+
+
+TEST(TestMexBinPlugin, write_bin_data_no_file) {
+    const std::string horace_root{
+        Environment::get_env_variable(Environment::HORACE_ROOT, ".") };
+
+    std::string binary_file{ horace_root + "/_test/binary_write.bin" };
+    del_file(binary_file);
+
+    std::vector<char> test_data(10 * 36, 'a');
+
+    std::unique_ptr<bin_io_handler> my_writer(new bin_io_handler);
+    fileParameters file_info;
+    file_info.fileName = binary_file;
+    file_info.nbin_start_pos = 0;
+    file_info.pix_start_pos = 60;
+    file_info.run_id = 0;
+    file_info.total_NfileBins = 0;
+    file_info.pixel_width = 32;
+
+    my_writer->init(file_info); // initialize writer class to write data
+    my_writer->write_pixels(&test_data[0], 10);
```

Comment:
There are many magic numbers in this function. why `10`, `60`, 20`, `12`? None are explained.

Reply:
they are random acceptable numbers